src/library_wasm_worker.js

```diff
@@ -0,0 +1,391 @@
+{{{ (function() { global.captureModuleArg = function() { return MODULARIZE ? '' : 'self.Module=d;'; }; return null; })(); }}}
+{{{ (function() { global.instantiateModule = function() { return MODULARIZE ? `${EXPORT_NAME}(d);` : ''; }; return null; })(); }}}
+{{{ (function() { global.instantiateWasm = function() { return MINIMAL_RUNTIME ? '' : 'd[`instantiateWasm`]=(i,r)=>{var n=new WebAssembly.Instance(d[`wasm`],i);r(n,d[`wasm`]);return n.exports};'; }; return null; })(); }}}
+
+#if WASM_WORKERS
+
+#if !SHARED_MEMORY
+#error Internal error! SHARED_MEMORY should be enabled when building with WASM_WORKERS
+#endif
+#if SINGLE_FILE
+#error -sSINGLE_FILE is not supported with -sWASM_WORKERS!
+#endif
+#if LINKABLE
+#error -sLINKABLE is not supported with -sWASM_WORKERS!
+#endif
+#if SIDE_MODULE
+#error -sSIDE_MODULE is not supported with -sWASM_WORKERS!
+#endif
+#if MAIN_MODULE
+#error -sMAIN_MODULE is not supported with -sWASM_WORKERS!
+#endif
+#if PROXY_TO_WORKER
+#error -sPROXY_TO_WORKER is not supported with -sWASM_WORKERS!
+#endif
+
+#endif // ~WASM_WORKERS
+
+
+mergeInto(LibraryManager.library, {
+  wasm_workers: {},
+  wasm_workers_id: 1,
+
+  // Starting up a Wasm Worker is an asynchronous operation, hence if the parent thread performs any
+  // postMessage()-based wasm function calls s to the Worker, they must be delayed until the async
+  // startup has finished, after which these postponed function calls can be dispatched.
+  _wasm_worker_delayedMessageQueue: [],
+
+  _wasm_worker_appendToQueue: function(e) {
+    __wasm_worker_delayedMessageQueue.push(e);
+  },
+
+  // Executes a wasm function call received via a postMessage.
+  _wasm_worker_runPostMessage: function(e) {
```

Comment:
Is postmessaging anything besides a proxied function to a Wasm Worker supported? It looks like this will unconditionally try to execute all messages as proxied functions.

Reply:
Yes, users can postMessage on their own, and this function will not react to the user provided postMessages. Only the messages that contain a member field `_wsc` will be attempted to be executed by this function.